apps/emqx_bridge/include/emqx_bridge_v2_register.hrl

```diff
@@ -0,0 +1,169 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2020-2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+%% This function is called to register a bridge V2. It should be called before
+%% the system boots in a function triggered by an -on_load() directive
+%% since it should be called before the system boots because the config
+%% system depends on that.
+%%
+%% It is placed in an hrl file instead of in emqx_bridge_v2.erl because emqx_bridge_v2
+%% might not be loaded when the bridge module is loaded.
+-spec emqx_bridge_v2_register_bridge_type(#{
+    %% Should be provided by all bridges. Even if the bridge_v2_type_name is
+    %% the same as the bridge_v1_type_named.
+    'bridge_v1_type_name' := atom(),
+    'bridge_v2_type_name' := atom(),
+    'connector_type' := atom(),
+    'schema_module' := atom(),
+    'schema_struct_field' := atom() | binary()
+}) -> ok.
```

Comment:
how about just a module, and everything else should be returned by the modules APIs
```suggestion
-spec emqx_bridge_v2_register_module(module()) -> ok.
```

Reply:
Good suggestion. Fixed! Rebased due to conflicts. The three last commits contains the real changes.